__tests__/cookie.test.js

```diff
@@ -269,25 +462,55 @@ describe("Basic user flow for Website", () => {
     expect(cookieButtonDisabled).toBe(false);
   }, 5000);
 
-  // Check to make sure that the speech synthesis is being populated, the correct voices are called by the function
-  // describe("populateVoice", () => {
-  //   test("populate voice options", () => {
-  //     const voices = [
-  //       { name: "Karen", lang: "en-AU", default: false },
-  //       { name: "Eddy", lang: "pt-BR", default: false },
-  //       { name: "Lesya", lang: "uk-UA", default: false },
-  //     ];
-  //     const synthesis = {
-  //       getVoices: jest.fn().mockReturnValue(voices),
-  //     };
-  //     //const option = document.createElement('option');
-  //     const appendchild = jest.spyOn(voiceSelect, "appendChild");
-  //     //document.body.innerHTML=
-  //     //<select></select>;
-  //     populateVoiceList();
-  //     expect(synthesis.getVoices).toHaveBeenCalled();
-  //     expect(appendchild).toHaveBeenCalledTimes(3);
-  //     //expect(voiceSelect.innerHTML).toContain('Karen');
-  //   });
-  // });
+  // Check to make sure that cancel button is not visible and has correct classes after fortune/cookie button are enabled
+  it("Make sure cancel button is not visible and has correct classes after fortune/cookie button are enabled", async () => {
+    console.log(
+      "Checking to make sure cancel button is visible and has correct classes after fortune/cookie button are enabled..."
+    );
+
+    // Only get past here if cancel button is visible
+    await page.waitForSelector("#cancel-animation-btn", { visible: false });
+
+    // cancel button should not have animating class
+    let elementHasClass = await page.evaluate(() => {
+      const cancelButton = document.querySelector("#cancel-animation-btn");
+      return cancelButton.parentElement.classList.contains("animating");
+    });
+    expect(elementHasClass).toBe(false);
+
+    // cancel button should have cancel-animation-wrapper class
+    elementHasClass = await page.evaluate(() => {
+      const cancelButton = document.querySelector("#cancel-animation-btn");
+      return cancelButton.parentElement.classList.contains(
+        "animating-new-cookie"
+      );
+    });
+    expect(elementHasClass).toBe(true);
+  }, 5000);
+
+  //Check to make sure that the speech synthesis is being populated, the correct voices are called by the function
+  it("Make sure that the speech synthesis options are working correctly", async() => {
+    console.log("check to make sure that speech synthesis list is working..");
+    //mock voice data
+    const voice = [
+      {name : 'Karen',lang: 'en-AU',default:false}
+    ];
+    window.speechSynthesis.getVoices.mockReturnValue(voice);
```

Comment:
Hmmm doesn't this need to be in evaluate or something, in order to access `window`

Reply:
I'll ask Do heon about it

Reply:
Should be resolved as of 7e947e9c70ab4406a2b8075a520d4a814b902d44